htsohm/material_files.py

```diff
@@ -197,131 +179,95 @@ def write_child_definition_files(run_id, parent_id, generation, mutation_strengt
+    new_number_density = (old_number_density + mutation_strength * (
+            random_number_density - old_number_density))
+    new_LCs = child_material.lattice_constants
+    new_vol = new_LCs['a'] * new_LCs['b'] * new_LCs['c']
+    number_of_atom_sites = int(new_number_density * new_vol)
+
     ########################################################################
     # remove excess atom-sites, if any
-    if number_of_atoms < len(old_x):
-        difference = len(old_x) - number_of_atoms
-        old_x = old_x[:-difference]
-        old_y = old_y[:-difference]
-        old_z = old_z[:-difference]
+    if number_of_atom_sites < len(parent_material.atom_sites):
+        parent_material.atom_sites = parent_material.atom_sites[
+                :number_of_atom_sites]
     ########################################################################
     # perturb atom-site positions
-    atom_sites = []
-    for i in range(len(old_x)):
-        chemical_id = old_atom_types[i]
-        x = random_position(old_x[i], random(), mutation_strength)
-        y = random_position(old_y[i], random(), mutation_strength)
-        z = random_position(old_z[i], random(), mutation_strength)
-        atom_site = {
-            "chemical-id" : chemical_id,
-            "x-frac"      : x,
-            "y-frac"      : y,
-            "z-frac"      : z}
-        atom_sites.append(atom_site)
+    child_material.atom_sites = []
+    for atom_site in parent_material.atom_sites:
+        new_atom_site = {'chemical-id' : atom_site}
+        for i in ['x-frac', 'y-frac', 'z-frac']
+            new_atom_site[i] = random_position(
+                    atom_site[i], random(), mutation_strength)
+        child_material.atom_sites.append(new_atom_site)
+
     ########################################################################
     # add atom-sites, if needed
-    if number_of_atoms > len(atom_sites):
+    if number_of_atom_sites > len(child_material.atom_sites):
         for new_sites in range(number_of_atoms - len(atom_sites)):
-            atom_site = {
-                "chemical-id" : choice(chemical_ids),
-                "x-frac"      : round(random(), 4),
-                "y-frac"      : round(random(), 4),
-                "z-frac"      : round(random(), 4)}
-            atom_sites.append(atom_site)
-
-    cif_file = os.path.join(mat_dir, run_id + '-' + str(new_material.uuid) + '.cif')
-    write_cif_file(cif_file, lattice_constants, atom_sites)
-
-    return new_material
+            new_atom_site = {
+                    'chemical-id' : choice(
+                        child_material.atom_types)['chemical-id']}
+            for i in ['x-frac', 'y-frac', 'z-frac']:
+                new_atom_site[i] = round(random(), 4)
+            atom_sites.append(new_atom_site)
+
+    material_file = os.path.join(material_dir, '%s.yaml' % uuid)
+    with open(material_file) as dump_file:
+        yaml.dump(child_material, dump_file) 
```

Comment:
We might consider sticking the yaml save on your PseudoMaterial class so the the PsuedoMaterial class knows how to save and load itself.